account_invoice_ubl_peppol/models/account_move.py

```diff
@@ -0,0 +1,404 @@
+# Copyright 2021 Sunflower IT (<https://sunflowerweb.nl>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from lxml import etree
+
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _account_invoice_ubl_use_peppol(self):
+        """ Returns True when we should use PEPPOL """
+        domain = self.company_id.get_ubl_domain_peppol()
+        return bool(self.filtered_domain(domain))
+
+    def generate_ubl_xml_string(self, version="2.1"):
+        self.ensure_one()
+        if self._account_invoice_ubl_use_peppol():
+            this = self.with_context(account_invoice_ubl_use_peppol=True)
```

Comment:
Shouldn't be better to add this to a field computed in post and store it on the invoice? 

With the current configuration, generation of the ubl is not consistent over time, as it depends on some rare configuration defined on a domain field inside the company. If we change this value, the generated document will change.

Also, I don't understand the need of the domain. Which is the standard use? Only for some journals or customers?

Reply:
I agree with your point that this should become a stored computed field.

My idea with the domain was that I assumed that the current UBL implementation was being used by someone somewhere, who made it that way for a reason, eg because it's how UBL should be done in France, and in come I with my PEPPOL hacking and slashing, and perhaps that's good for NL invoices but not for all countries. So the main idea was for it to be a method whereby people can say hey, we use this for NL, or even for whole EU, but for the other invoices we use the normal `_ubl` implementation.

However, earlier during review Alexis already kicked in and said that there wasn't any specific scheme used and that we might as well make PEPPOL the default implementation and let any exceptions identify themselves as child modules that ADD something, instead of now the PEPPOL module mostly REMOVING things. But he suggested doing that in the 14.0+ version.

I can refactor, but we can also leave that for the 14.0+ version. Is this blocking for you?